### PR TITLE
hardening/1314_refactor_carto_historical_mode_names

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #1314 
* Relevant unit tests:
```
[NGSICartoDBSink.configure] ----------------------------------------- Only enable_raw configuration works
[NGSICartoDBSink.configure] ----------------------------------  OK  - Only 'enable_raw' was configured and worked
[NGSICartoDBSink.configure] ----------------------------------------- Only enable_distance configuration works
[NGSICartoDBSink.configure] ----------------------------------  OK  - Only 'enable_distance' was configured and worked
[NGSICartoDBSink.configure] ----------------------------------------- When enable_raw and enable_raw_historic are configured, the second one value is used
[NGSICartoDBSink.configure] ----------------------------------  OK  - Both 'enable_raw' and 'enable_raw_historic' where configured, but 'enable_raw_historic' value is used
[NGSICartoDBSink.configure] ----------------------------------------- When enable_distance and enable_distance_historic are configured, the second one value is used
[NGSICartoDBSink.configure] ----------------------------------  OK  - Both 'enable_distance' and 'enable_distance_historic' where configured, but 'enable_distance_historic' value is used
```

### CODE REVIEW

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 1]
```diff
@@ -446,8 +448,8 @@ cygnus-ngsi.sinks.cartodb-sink.enable_name_mappings = false
 cygnus-ngsi.sinks.cartodb-sink.enable_lowercase = false
 cygnus-ngsi.sinks.cartodb-sink.keys_conf_file = /usr/cygnus/conf/cartodb_keys.conf
 cygnus-ngsi.sinks.cartodb-sink.swap_coordinates = true
-cygnus-ngsi.sinks.cartodb-sink.enable_raw = true
-cygnus-ngsi.sinks.cartodb-sink.enable_distance = false
+cygnus-ngsi.sinks.cartodb-sink.enable_raw_historic = true
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: A change in the configuration use to break backward compatibility... are the old ones still supported? Has been them mentioned in the deprecated documentation (if Cygnus has such kind of documentation?

Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Yes, everything is within this PR :) Both old and new parameters are supported, nevertheless old parameters warn about they are deprecated when used. In addition, the deprecated functinality document has nen entries for this.

Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Alfter looking the whole PR I have realized that is supported and is mentioned in deprecated documenation. However, an explicit confirmation will be fine :)

Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**